python_aws_ssm/parameters.py

```diff
@@ -31,11 +41,20 @@ def get_parameters(self, ssm_key_names: List[str]) -> Dict[str, Optional[str]]:
             have a matching value of None.
         If SSM somehow returns keys that are not requested, these keys are not
         returned in the result dict.
+
+        :raises: InvalidParametersError when invalid parameters were requested.
         """
 
-        retrieved_parameters = self.client.get_parameters(
+        response: Dict = self.client.get_parameters(
             Names=ssm_key_names, WithDecryption=True
-        ).get("Parameters")
+        )
+        if (
+            response.get("InvalidParameters")
+            and type(response["InvalidParameters"]) is list
```

Comment:
this seems a bit confusing, but I didn't check the client response. Is there a case where we have `InvalidParameters` with a different type than a list?

Reply:
That's a fair question. I haven't seen a case like that so maybe this is overly defensive. Should I change it to `if response.get("InvalidParameters"):`?

Reply:
it's probably better 👍

Reply:
Fixed 👍